mmv1/products/looker/Instance.yaml

```diff
@@ -0,0 +1,368 @@
+    name: maintenanceWindow
+    description: |
+      Maintenance window for an instance.
+
+      Maintenance of your instance takes place once a month, and will require
+      your instance to be restarted during updates, which will temporarily
+      disrupt service.
+    properties:
+      - !ruby/object:Api::Type::Enum
+        name: 'dayOfWeek'
+        required: true
+        description: |
+          Required. Day of the week for this MaintenanceWindow (in UTC).
+
+          - DAY_OF_WEEK_UNSPECIFIED: The day of the week is unspecified.
+          - MONDAY: Monday
+          - TUESDAY: Tuesday
+          - WEDNESDAY: Wednesday
+          - THURSDAY: Thursday
+          - FRIDAY: Friday
+          - SATURDAY: Saturday
+          - SUNDAY: Sunday
+        values:
+          - :DAY_OF_WEEK_UNSPECIFIED
+          - :MONDAY
+          - :TUESDAY
+          - :WEDNESDAY
+          - :THURSDAY
+          - :FRIDAY
+          - :SATURDAY
+          - :SUNDAY
+      - !ruby/object:Api::Type::NestedObject
+        name: 'startTime'
+        required: true
+        description: |
+          Required. Start time of the window in UTC time.
+        properties:
+          - !ruby/object:Api::Type::Integer
+            name: 'hours'
+            description: |
+              Hours of day in 24 hour format. Should be from 0 to 23.
+            validation: !ruby/object:Provider::Terraform::Validation
+              function: 'validation.IntBetween(0,23)'
+          - !ruby/object:Api::Type::Integer
+            name: 'minutes'
+            description: |
+              Minutes of hour of day. Must be from 0 to 59.
+            validation: !ruby/object:Provider::Terraform::Validation
+              function: 'validation.IntBetween(0,60)'
+          - !ruby/object:Api::Type::Integer
+            name: 'seconds'
+            description: |
+              Seconds of minutes of the time. Must normally be from 0 to 59.
+            validation: !ruby/object:Provider::Terraform::Validation
+              function: 'validation.IntBetween(0,60)'
+          - !ruby/object:Api::Type::Integer
+            name: 'nanos'
+            description: |
+              Fractions of seconds in nanoseconds. Must be from 0 to 999,999,999.
+            validation: !ruby/object:Provider::Terraform::Validation
+              function: 'validation.IntBetween(0,999999999)'
+  # Maintenance Window Object - End
+  - !ruby/object:Api::Type::String
+    name: name
+    description: |
+      The ID of the instance or a fully qualified identifier for the instance.
+    required: true
+    immutable: true
+    url_param_only: true
+    validation: !ruby/object:Provider::Terraform::Validation
+      regex: '^[a-z][a-z0-9-]{0,39}[a-z0-9]$'
+  # Oauth Object
+  - !ruby/object:Api::Type::NestedObject
+    name: oauthConfig
+    ignore_read: true
+    description: |
+      Looker Instance OAuth login settings.
+    properties:
+      - !ruby/object:Api::Type::String
+        name: 'clientId'
```

Comment:
Should these values be required if `oauthConfig` is set? Right now a user could specify any of:

```
oauth_config {}
```

```
oauth_config {
  client_id = "value"
}
```

```
oauth_config {
  client_secret = "value"
}
```

Reply:
Done.